enterprise/backend/src/metabase_enterprise/transforms/api.clj

```diff
@@ -55,14 +66,17 @@
              :schema "transforms"
              :name "gadget_products"}}])
 
-(defn- source-database-id
+;; TODO this and target-database-id can be transforms multimethods?
+(defn- target-database-id
   [transform]
-  (-> transform :source :query :database))
+  (if (transforms.util/python-transform? transform)
+    (-> transform :target :database)
+    (-> transform :source :query :database)))
```

Comment:
we should at least move this to `util`, and use it in `target-table-exists?` and `delete-target-table!`. we can make it a multimethod later if needed

Reply:
yea, i really want this to be a transform multimethod, but let's do that next week